src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/RealWorldContextTests.cs

```diff
@@ -149,37 +149,44 @@ public void RoundtripJsonElement(string json)
         [Fact]
         public virtual void RoundTripValueTuple()
         {
-            bool isIncludeFieldsEnabled = DefaultContext.IsIncludeFieldsEnabled;
-
             var tuple = (Label1: "string", Label2: 42, true);
-            string expectedJson = isIncludeFieldsEnabled
-                ? "{\"Item1\":\"string\",\"Item2\":42,\"Item3\":true}"
-                : "{}";
+            // Tuples now always serialize as objects with Item1, Item2, Item3 regardless of IncludeFields
+            string expectedJson = "{\"Item1\":\"string\",\"Item2\":42,\"Item3\":true}";
 
             string json = JsonSerializer.Serialize(tuple, DefaultContext.ValueTupleStringInt32Boolean);
             Assert.Equal(expectedJson, json);
 
             if (DefaultContext.JsonSourceGenerationMode == JsonSourceGenerationMode.Serialization)
             {
                 // Deserialization not supported in fast path serialization only mode
-                // but if there are no fields we won't throw because we throw on the property lookup
-                if (isIncludeFieldsEnabled)
-                {
-                    Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize(json, DefaultContext.ValueTupleStringInt32Boolean));
-                }
-                else
-                {
-                    (string, int, bool) obj = JsonSerializer.Deserialize(json, DefaultContext.ValueTupleStringInt32Boolean);
-                    Assert.Equal(default((string, int, bool)), obj);
-                }
+                Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize(json, DefaultContext.ValueTupleStringInt32Boolean));
             }
             else
             {
                 var deserializedTuple = JsonSerializer.Deserialize(json, DefaultContext.ValueTupleStringInt32Boolean);
-                Assert.Equal(isIncludeFieldsEnabled ? tuple : default, deserializedTuple);
+                Assert.Equal(tuple, deserializedTuple);
             }
         }
 
+        [Fact]
+        public virtual void LongValueTupleSerializes()
+        {
+            // Test 8-element tuple (boundary case)
+            var tuple8 = (1, 2, 3, 4, 5, 6, 7, 8);
+            string json8 = JsonSerializer.Serialize(tuple8);
+            Assert.Equal("{\"Item1\":1,\"Item2\":2,\"Item3\":3,\"Item4\":4,\"Item5\":5,\"Item6\":6,\"Item7\":7,\"Item8\":8}", json8);
```

Comment:
Please also test deserialization. Because tuples have structural equality you can simply Assert.Equals with the original value.